pkg/api/unversioned.go

```diff
@@ -22,16 +22,64 @@ import (
 
 // This file contains API types that are unversioned.
 
-// APIVersions lists the api versions that are available, to allow
-// version negotiation. APIVersions isn't just an unnamed array of
-// strings in order to allow for future evolution, though unversioned
+// APIVersions lists the versions that are available, to allow clients to
+// discover the API at /api.
 type APIVersions struct {
+	// Versions are the api versions that are available.
 	Versions []string `json:"versions"`
 }
 
+// APIGroupList is a list of APIGroup, to allow clients to discover the API at
+// /apis.
+type APIGroupList struct {
+	// Groups is a list of APIGroup.
+	Groups []APIGroup `json:"groups"`
+}
+
+// APIGroup contains the name, the supported versions, and the preferred version
+// of a group.
+type APIGroup struct {
+	// GroupName represents the name of the group.
+	GroupName string `json:"groupName"`
```

Comment:
Our convention is to take advantage of context. Since this is an APIGroup, "group" is redundant in the field name. If we were to stick to this convention, this field would be called simply `name`. Did someone feel that was ambiguous in this situation? If so, was it in part because this type doesn't have a `kind` field?